src/python/pants/engine/legacy/graph.py

```diff
@@ -254,6 +245,25 @@ def _inject(self, subjects):
           yielded_addresses.add(address)
           yield address
 
+  def _assert_type_is_return(self, node, state):
```

Comment:
As part of #4229, we should remove the need to check the return type of exceptions. Maybe include a TODO here pointing to that one.